File: myintent/src/main/java/pl/mareklangiewicz/myintent/MICmdListFragment.java
package pl.mareklangiewicz.myintent;

import android.os.Bundle;
import android.support.annotation.DrawableRes;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import pl.mareklangiewicz.myfragments.MyFragment;

/**
 * Created by Marek Langiewicz on 12.11.15.
 */
public class MICmdListFragment extends MyFragment {

    private @Nullable RecyclerView mRecyclerView;
    private final CmdAdapter mAdapter = new CmdAdapter();

    @Nullable @Override public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        super.onCreateView(inflater, container, savedInstanceState); // just for logging

        View rootView = inflater.inflate(R.layout.mi_cmd_list_fragment, container, false);

        mRecyclerView = (RecyclerView) rootView.findViewById(R.id.mi_cmd_recycler_view);
        mRecyclerView.setHasFixedSize(true);
        LinearLayoutManager manager = new LinearLayoutManager(getActivity());
        //noinspection ConstantConditions
        mRecyclerView.setLayoutManager(manager);
        mRecyclerView.setAdapter(mAdapter);

        return rootView;
    }

    @Override
    public void onDestroyView() {
        //noinspection ConstantConditions
        mRecyclerView.setAdapter(null);
        mRecyclerView = null;
        super.onDestroyView();
    }

    public void setCommands(List<String> commands) {
        mAdapter.setCommands(commands);
    }
    public void setImageRes(@DrawableRes int res) { mAdapter.setImageRes(res); }

    public @Nullable List<String> getCommands() { return mAdapter.getCommands(); }
}

File: myintent/src/main/java/pl/mareklangiewicz/myintent/MICmdListFragment.kt
package pl.mareklangiewicz.myintent

import android.os.Bundle
import android.support.annotation.DrawableRes
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.mi_cmd_list_fragment.*
import pl.mareklangiewicz.myfragments.MyFragment

/**
 * Created by Marek Langiewicz on 12.11.15.
 */
open class MICmdListFragment : MyFragment() {

    private val mAdapter = CmdAdapter()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState) // just for logging
        return inflater.inflate(R.layout.mi_cmd_list_fragment, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mi_cmd_recycler_view.setHasFixedSize(true)
        mi_cmd_recycler_view.layoutManager = LinearLayoutManager(activity)
        mi_cmd_recycler_view.adapter = mAdapter
    }

    fun setImageRes(@DrawableRes res: Int) {
        mAdapter.imageRes = res
    }

    var commands: List<String>?
        get() {
            return mAdapter.commands
        }
        set(commands) {
            mAdapter.commands = commands
        }
}

